tools/deviceGen/input/device_features.json

```diff
@@ -0,0 +1,2666 @@
+        },
+        {
+            "type": "bool",
+            "name": "pipelineStatisticsQuery",
+            "value": false
+        },
+        {
+            "groupComment": [
+                "[EXPOSE AS LIMIT] Enabled by Default, Moved to Limits"
+            ]
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "vertexPipelineStoresAndAtomics"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "fragmentStoresAndAtomics"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "shaderTessellationAndGeometryPointSize"
+        },
+        {
+            "type": "bool",
+            "name": "shaderImageGatherExtended",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] ROADMAP 2024 good device support"
+        },
+        {
+            "type": "bool",
+            "name": "shaderStorageImageExtendedFormats",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] ROADMAP 2024 good device support"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "shaderStorageImageMultisample",
+            "comment": "[EXPOSE AS LIMIT] Cannot be always enabled cause Intel ARC is handicapped"
+        },
+        {
+            "groupComment": [
+                "TODO: format feature reporting unimplemented yet for both of the below! (should we move to usage reporting?)",
+                "[EXPOSE AS LIMIT] always enable, shouldn't cause overhead by just being enabled"
+            ]
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "shaderStorageImageReadWithoutFormat"
+        },
+        {
+            "type": "bool",
+            "name": "shaderStorageImageWriteWithoutFormat",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] good device support"
+        },
+        {
+            "groupComment": [
+                "[REQUIRE] ROADMAP 2022 and good device support"
+            ]
+        },
+        {
+            "type": "bool",
+            "name": "shaderUniformBufferArrayDynamicIndexing",
+            "value": true,
+            "expose": false
+        },
+        {
+            "type": "bool",
+            "name": "shaderSampledImageArrayDynamicIndexing",
+            "value": true,
+            "expose": false
+        },
+        {
+            "type": "bool",
+            "name": "shaderStorageBufferArrayDynamicIndexing",
+            "value": true,
+            "expose": false
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "shaderStorageImageArrayDynamicIndexing",
+            "comment": "[EXPOSE AS A LIMIT] ROADMAP 2022 but Apple GPUs have poor support"
+        },
```

Comment:
again they should be in a block under `groupComment`

Reply:
except `shaderStorageImageArrayDynamicIndexing` cause thats not in the group

Reply:
Done